scylla/src/network/connection_pool.rs

```diff
@@ -1187,6 +1187,16 @@ struct OpenedConnectionEvent {
     keyspace_name: Option<VerifiedKeyspaceName>,
 }
 
+/// Signals that connectivity to a node has changed.
+#[derive(Debug)]
+pub(crate) enum ConnectivityChangeEvent {
+    /// A new connection to the node was established, while there were no working connections.
+    Established { address: NodeAddr },
+
+    /// The last working connection to the node was lost.
+    Lost { address: NodeAddr },
+}
+
```

Comment:
Commit: " connection_pool: introduce ConnectivityChangeEvent "

It is surprising that NodeAddr is used for identification, given all the issues and ambiguities we have with identifying nodes this way. Maybe the reason for that will become clear in later commits, idk.

Reply:
I use it because this is an internal API, so I preferred to keep the additional information that NodeAddr has compared to SocketAddr. It's up to the upper layer to convert it to SocketAddr, which it btw does.

Reply:
Shall we consider this resolved?